### PR TITLE
Add table oci release notes

### PR DESCRIPTION
**Pull request by lucasjellema**
Fix for issue #486 

# Integration test logs
<details>
  <summary>Logs</summary>
itpod /workspace/steampipe-plugin-oci/oci-test (add-table-oci-release-notes) $ node ./tint.js oci_releasenote
Test names: [ 'tests/oci_releasenote' ]
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging

SETUP: tests/oci_releasenote []

PRETEST: tests/oci_releasenote

TEST: tests/oci_releasenote
Running terraform

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration
and found no differences, so no changes are needed.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Running SQL query: test-list-query.sql

[
  {
    "release_date": "2023-03-02T00:00:00Z",
    "title": "Speech service available in US Government Cloud with FedRAMP Authorization"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "release_date": "2023-03-02T00:00:00Z",
    "title": "Speech service available in US Government Cloud with FedRAMP Authorization"
  }
]
✔ PASSED

POSTTEST: tests/oci_releasenote

TEARDOWN: tests/oci_releasenote

SUMMARY:

1/1 passed.
</details>

# Example query results
<details>
  <summary>Results</summary>
gitpod /workspace/steampipe-plugin-oci/oci-test (add-table-oci-release-notes) $ steampipe query "select 
>   title,
>   service, 
>   all_services,
>   summary, 
>   release_date 
> from 
>   oci_releasenote 
> limit 10 "

```+-----------------------------------------------------------------------------+----------------------------------------------+-----------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+----------------------+
| title                                                                       | service                                      | all_services                                              | summary                                                                                                                                    | release_date         |
+-----------------------------------------------------------------------------+----------------------------------------------+-----------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+----------------------+
| Data Safe: Advanced filtering using SCIM in audit and alert reports         | Data Safe                                    | ["Data Safe"]                                             | You can now add advanced filters to your audit or alerts reports.                                                                          | 2023-02-23T00:00:00Z |
| Data Safe: Support for Autonomous Database on Oracle Exadata Cloud@Customer | Data Safe                                    | ["Data Safe"]                                             | You can now register an Autonomous Database on Oracle Exadata Cloud@Customer as an Oracle Data Safe target database.                       | 2023-02-23T00:00:00Z |
| Data Flow Supports Arm-Based Compute Shapes                                 | Data Flow                                    | ["Data Flow"]                                             | You can now select Arm-based compute shapes for your drivers and executors with Data Flow. For more information, see the …                 | 2023-03-01T00:00:00Z |
| Data Safe: US Government Cloud regions are now supported                    | Data Safe                                    | ["Data Safe"]                                             | Oracle Data Safe is now supported in the US Gov West (Phoenix) and US Gov East (Ashburn) regions.                                          | 2023-02-23T00:00:00Z |
| Synthetic Monitoring Enhancements                                           | Application Performance Monitoring           | ["Application Performance Monitoring"]                    | A new release is available with the following Synthetic Monitoring features. You can now:                                                  | 2023-02-28T00:00:00Z |
| Speech service available in US Government Cloud with FedRAMP Authorization  | Oracle Cloud Infrastructure Government Cloud | ["Oracle Cloud Infrastructure Government Cloud","Speech"] | Speech is now available in the US Government Cloud with FedRAMP Authorization.For more information about this service in the …             | 2023-03-02T00:00:00Z |
| Custom Query Bars in Trace Explorer                                         | Application Performance Monitoring           | ["Application Performance Monitoring"]                    | You can create and save Custom Query Bars for later use, and edit saved query bars from the new Open …                                     | 2023-02-28T00:00:00Z |
| X9-based virtual machine GPU shapes for Compute instances                   | Compute                                      | ["Compute"]                                               | Two new Intel X9-based GPU shapes are now available for virtual machine instances.                                                         | 0001-01-01T00:00:00Z |
| Oracle Autonomous Linux 8 now available as a platform image                 | Compute                                      | ["Compute"]                                               | Oracle Autonomous Linux 8 is now available as a platform image. For more information, see Platform Images.                                 | 2023-02-23T00:00:00Z |
| APM: OpenTelemetry Native Protocol is Now Supported                         | Application Performance Monitoring           | ["Application Performance Monitoring"]                    | You can now configure OpenTelemetry data sources to upload trace and metric data using the native OpenTelemetry protocol (OTLP) directly … | 2023-02-28T00:00:00Z |
+-----------------------------------------------------------------------------+----------------------------------------------+-----------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+----------------------+
```
gitpod /workspace/steampipe-plugin-oci/oci-test (add-table-oci-release-notes) $ steampipe query "select 
>   title,
>   service, 
>   summary, 
>   release_date 
> from 
>   oci_releasenote 
> where 
>   release_date  > current_date - interval '1' month;
> "

```+--------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+
| title                                                                                                                                | service                                      | summary                                                                                                                                             | release_date         |
+--------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+
| New Available Region and Additional Capabilities Introduced in Oracle Access Governance                                              | Access Governance                            | The following new capabilities are introduced in Oracle Access Governance:                                                                          | 2023-02-23T00:00:00Z |
| Data Flow Supports Arm-Based Compute Shapes                                                                                          | Data Flow                                    | You can now select Arm-based compute shapes for your drivers and executors with Data Flow. For more information, see the …                          | 2023-03-01T00:00:00Z |
| Speech service available in US Government Cloud with FedRAMP Authorization                                                           | Oracle Cloud Infrastructure Government Cloud | Speech is now available in the US Government Cloud with FedRAMP Authorization.For more information about this service in the …                      | 2023-03-02T00:00:00Z |
| Custom Query Bars in Trace Explorer                                                                                                  | Application Performance Monitoring           | You can create and save Custom Query Bars for later use, and edit saved query bars from the new Open …                                              | 2023-02-28T00:00:00Z |
| Synthetic Monitoring Enhancements                                                                                                    | Application Performance Monitoring           | A new release is available with the following Synthetic Monitoring features. You can now:                                                           | 2023-02-28T00:00:00Z |
| Data Safe: US Government Cloud regions are now supported                                                                             | Data Safe                                    | Oracle Data Safe is now supported in the US Gov West (Phoenix) and US Gov East (Ashburn) regions.                                                   | 2023-02-23T00:00:00Z |
| Data Safe: Audit Insights                                                                                                            | Data Safe                                    | You can now use Audit Insights to view and analyze the audit volume of several items, such as target databases, schemas, and database users.        | 2023-02-23T00:00:00Z |
| Oracle Autonomous Linux 8 now available as a platform image                                                                          | Compute                                      | Oracle Autonomous Linux 8 is now available as a platform image. For more information, see Platform Images.                                          | 2023-02-23T00:00:00Z |
| Discover and Monitor External Database Systems in Database Management                                                                | Database Management                          | You can now discover an External Database System, which is an Oracle Database System located outside of Oracle Cloud Infrastructure, …              | 2023-02-23T00:00:00Z |
| APM: OpenTelemetry Native Protocol is Now Supported                                                                                  | Application Performance Monitoring           | You can now configure OpenTelemetry data sources to upload trace and metric data using the native OpenTelemetry protocol (OTLP) directly …          | 2023-02-28T00:00:00Z |
| Data Safe: Advanced filtering using SCIM in audit and alert reports                                                                  | Data Safe                                    | You can now add advanced filters to your audit or alerts reports.                                                                                   | 2023-02-23T00:00:00Z |
| Manage master encryption key wallets in OCI GoldenGate                                                                               | GoldenGate                                   | You can now manage master encryption key wallets in OCI GoldenGate. You can use master encryption keys to encrypt trail …                           | 2023-02-07T00:00:00Z |
| New Release for Data Integration (fixes)                                                                                             | Data Integration                             | This release contains the following fixes.                                                                                                          | 2023-02-19T00:00:00Z |
| Data Safe: Support for Autonomous Database on Oracle Exadata Cloud@Customer                                                          | Data Safe                                    | You can now register an Autonomous Database on Oracle Exadata Cloud@Customer as an Oracle Data Safe target database.                                | 2023-02-23T00:00:00Z |
| File Storage Cross-Region Replication is Now Available in US Government Cloud                                                        | File Storage                                 | File System Replication is now available in the US Government Cloud. Cross-region replication provides protection from regional outages, aids …     | 2023-02-09T00:00:00Z |
| UI-based Discovery for PeopleSoft Resources                                                                                          | Stack Monitoring                             | Users can now perform PeopleSoft resource discovery directly from the Stack Monitoring UI, greatly simplifying the discovery process.For more …     | 2023-02-09T00:00:00Z |
| Visualize Synthetic Monitor Runs in Network Data Explorer                                                                            | Application Performance Monitoring           | You can now visualize the end-to-end network data path of monitor runs on one or more vantage points in the …                                       | 2023-02-08T00:00:00Z |
| Web Application Firewall available in US Government Cloud with FedRAMP Authorization                                                 | Oracle Cloud Infrastructure Government Cloud | Web Application Firewall is now available in the US Government Cloud with FedRAMP Authorization.For more information about this service …           | 2023-02-06T00:00:00Z |
| Data Safe: New Activity Auditing reports                                                                                             | Data Safe                                    | Additional Activity Auditing reports have been added to Oracle Data Safe.                                                                           | 2023-02-23T00:00:00Z |
| Oracle Database Autonomous Recovery Service is Now Available                                                                         | Oracle Cloud Infrastructure                  | Oracle Database Zero Data Loss Autonomous Recovery Service is a fully managed data protection service for Oracle databases running on …             | 2023-02-15T00:00:00Z |
| MySQL Database Service Support for MySQL Version 8.0.32                                                                              | MySQL Database                               | MySQL Database Service now supports MySQL 8.0.32. New MySQL DB systems are based on this latest MySQL database version. For …                       | 2023-02-08T00:00:00Z |
| Media Flow now supports SRT and JSON formats for creating subtitles                                                                  | Media Flow                                   | Oracle Cloud Infrastructure Media Flow now supports SRT and JSON formats for creating subtitles. For more information, see Media Flow …             | 2023-02-07T00:00:00Z |
| Artifact Registry available in US Government Cloud with FedRAMP Authorization                                                        | Artifact Registry                            | Artifact Registry is now available in the US Government Cloud with FedRAMP Authorization.For more information about this service in …               | 2023-02-13T00:00:00Z |
| Data Science notebook session JupyterLab interface enhancements                                                                      | Data Science                                 |                                                                                                                                                     | 2023-02-22T00:00:00Z |
| New release for Data Catalog                                                                                                         | Data Catalog                                 | A new release for Data Catalog is now available. Apart from bug fixes and enhancements, you can now:                                                | 2023-02-08T00:00:00Z |
| Anomaly Detection now includes Univariate Anomaly Detection, Multivariate Anomaly Detection improvements, and Asynchronous Detection | Anomaly Detection                            | We've added support for detecting anomalies in univariate signals that allow you to detect different types of anomalies in univariate …             | 2023-02-22T00:00:00Z |
| New protection capabilities are available for use with in-region WAF                                                                 | Web Application Firewall                     | The Web Application Firewall (WAF) has been updated to include new WAF protection capabilities. These protection capabilities include rules for …   | 2023-02-10T00:00:00Z |
| Oracle Visual Builder Studio is now available                                                                                        | Visual Builder Studio                        | Oracle Visual Builder Studio (VB Studio) is a robust application development platform that helps your team collaboratively develop, test, deploy, … | 2023-02-14T00:00:00Z |
| DevOps now supports validating the Helm chart                                                                                        | DevOps                                       | DevOps now supports verifying the integrity of Helm chart before the deployment. If the verification fails, then the Helm chart is …                | 2023-02-07T00:00:00Z |
+--------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+
```
gitpod /workspace/steampipe-plugin-oci/oci-test (add-table-oci-release-notes) $ steampipe query "select 
>   title,
>   summary, 
>   release_date,
>   all_services 
> from 
>   oci_releasenote 
> where 
>   all_services  ? 'Speech'"
```+----------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------+----------------------+-----------------------------------------------------------+
| title                                                                      | summary                                                                                                                        | release_date         | all_services                                              |
+----------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------+----------------------+-----------------------------------------------------------+
| Speech service available in US Government Cloud with FedRAMP Authorization | Speech is now available in the US Government Cloud with FedRAMP Authorization.For more information about this service in the … | 2023-03-02T00:00:00Z | ["Oracle Cloud Infrastructure Government Cloud","Speech"] |
| New Languages Supported for Speech                                         | The following new languages are supported with this release:                                                                   | 0001-01-01T00:00:00Z | ["Speech"]                                                |
| New features in Speech                                                     | You can now use these new features in Speech:                                                                                  | 2022-04-28T00:00:00Z | ["Speech"]                                                |
| Speech service is now available                                            | You can use the Speech service to convert audio files to readable text that is stored in JSON format. Speech harnesses the …   | 2022-02-23T00:00:00Z | ["Speech"]                                                |
+----------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------+----------------------+-----------------------------------------------------------+
```
gitpod /workspace/steampipe-plugin-oci/oci-test (add-table-oci-release-notes) $ steampipe query "select 
>   title,
>   summary, 
>   release_date,
>   all_services 
> from 
>   oci_releasenote 
> where 
>   service  = 'Anomaly Detection' 
>  "
```+--------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+--------------------------------------------------------------+
| title                                                                                                                                | summary                                                                                                                                                    | release_date         | all_services                                                 |
+--------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+--------------------------------------------------------------+
| Anomaly Detection is now available                                                                                                   | The Anomaly Detection service helps you to identify anomalies in a multivariate dataset by taking advantage of the interrelationship among signals. This … | 2021-07-13T00:00:00Z | ["Anomaly Detection"]                                        |
| Addition of new regions                                                                                                              | Anomaly Detection is now available in the following regions:For more information about Anomaly …                                                           | 0001-01-01T00:00:00Z | ["Anomaly Detection"]                                        |
| New Services available in US Government Cloud with FedRAMP Authorization                                                             | The following services are now available in the US Government Cloud with FedRAMP Authorization:For more …                                                  | 0001-01-01T00:00:00Z | ["Anomaly Detection","Billing","DevOps","Anomaly Detection"] |
| Anomaly Detection now includes Univariate Anomaly Detection, Multivariate Anomaly Detection improvements, and Asynchronous Detection | We've added support for detecting anomalies in univariate signals that allow you to detect different types of anomalies in univariate …                    | 2023-02-22T00:00:00Z | ["Anomaly Detection"]                                        |
| Addition of new region                                                                                                               | Anomaly Detection is now available in the following region:For more information about Anomaly Detection and features …                                     | 2022-11-08T00:00:00Z | ["Anomaly Detection"]                                        |
| Anomaly Detection now has multivariate and univariate                                                                                | The Anomaly Detection service is a multivariate anomaly detection solution that generates a model by analyzing multiple timeseries signal data …           | 2021-12-16T00:00:00Z | ["Anomaly Detection"]                                        |
| Addition of new regions                                                                                                              | Anomaly Detection is now available in these regions:                                                                                                       | 0001-01-01T00:00:00Z | ["Anomaly Detection"]                                        |
| Required policy and addition of Johannesburg region                                                                                  | A required policy is added as a prerequiste to using Anomaly Detection.Support for the Johannesburg region was added.You …                                 | 2022-01-28T00:00:00Z | ["Anomaly Detection"]                                        |
+--------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+--------------------------------------------------------------+
```
</details>

### CODE REVIEW

**Comment by cbruno10** [comment]
Hi @lucasjellema , thanks for opening this PR and also implementing the table in a clever way!

Are you able to share more on how you're looking to use this table? I'm not sure if this would help with your requirements, but I figured I would mention we have an RSS plugin that allows you to query RSS channel and item information for one or more feeds (we have some staged changes in https://github.com/turbot/steampipe-plugin-rss/tree/issue-24 to allow passing in feed links through the config).

We then use this plugin in our RSS Insights mod to create dashboards that allow us to create views like:
Screen Shot 2023-03-17 at 2 39 38 PM
(source available in https://github.com/turbot/steampipe-mod-rss-insights/blob/initial-branch/dashboards/rss_reader.sp).

In this PR, I can see you're pulling out service info and allowing users to pass in a `service` qual, which is not possible out of the box with the `rss_item` column, but could be created through a control or dashboard with inputs.

File: oci/table_oci_releasenote.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+// the url where the web pages with release notes are available
+const RELEASE_NOTES_BASE_URL = "https://docs.oracle.com/en-us/iaas/releasenotes/"
+const BASE_DOCUMENTATION_URL = "https://docs.oracle.com"
+
+func getOCIReleaseNotes(service string, page int) (releaseNotes []ReleaseNote, numberOfPages int, err error) {
+	c := colly.NewCollector(
+		colly.AllowedDomains(ALLOWED_DOMAIN),
+	)
+	releaseNotes = make([]ReleaseNote, 0, 50)
+	c.OnHTML(".uk-article", func(e *colly.HTMLElement) {
```
Comment (cbruno10) [inline oci/table_oci_releasenote.go:165]: In our RSS plugin, we use [gofeed](https://github.com/mmcdole/gofeed), which makes parsing RSS for that table much easier. Have you seen this package, would it help you parse the RSS feeds more cleanly?

Comment (cbruno10) [reply oci/table_oci_releasenote.go:165]: A few example uses in the plugin (we don't manipulate the response data that much, but instead just pass them to columns):
- https://github.com/turbot/steampipe-plugin-rss/blob/315ac92bf5b5fb49bf8ea449ad292d190b354ceb/rss/table_rss_channel.go#L46-L56
- https://github.com/turbot/steampipe-plugin-rss/blob/315ac92bf5b5fb49bf8ea449ad292d190b354ceb/rss/table_rss_item.go#L44-L60

**Comment by lucasjellema** [comment]
> Hi @lucasjellema , thanks for opening this PR and also implementing the table in a clever way!
> 
> Are you able to share more on how you're looking to use this table? I'm not sure if this would help with your requirements, but I figured I would mention we have an RSS plugin that allows you to query RSS channel and item information for one or more feeds (we have some staged changes in https://github.com/turbot/steampipe-plugin-rss/tree/issue-24 to allow passing in feed links through the config).
> 
> We then use this plugin in our RSS Insights mod to create dashboards that allow us to create views like: Screen Shot 2023-03-17 at 2 39 38 PM (source available in https://github.com/turbot/steampipe-mod-rss-insights/blob/initial-branch/dashboards/rss_reader.sp).
> 
> In this PR, I can see you're pulling out service info and allowing users to pass in a `service` qual, which is not possible out of the box with the `rss_item` column, but could be created through a control or dashboard with inputs.

This table does two things: it provides records for all release notes regarding OCI (since the beginning of time) as well as the subset of release notes for a particular service. Oracle provides an RSS feed for the first category only - and the contains only the most recent 50 items. In my implementation I make use of the web pages that Oracle publishes - one set of pages with all 3000+ notes and another set of pages that each cover all release notes for a particular service. Using the RSS plugin would only help with a fairly small portion of the data this table exposes.

I would like to use the this table myself to list/report on release notes - probably in a dashboard. For a generic overview, the 50 most recent notes are probably enough. For service specific overviews of the most recent changes, only drawing from the RSS feed definitely would not be enough (with over 30 services, most services do not appear more than once or even at all in the RSS feed).

**Comment by cbruno10** [comment]
Hey @lucasjellema , sorry for the long response! We've been going through open PRs in this repo, along with others, but have been moving through them a bit slower than we'd like.

For this PR at the moment, we don't have plans to accept it, as we'd like to actually create a plugin that returns scrapped data using the package you've used (or a similar one). If the plugin was designed correctly, we feel this approach would allow you to get the release note data you're getting in this table, but also be flexible enough to scrape other types of pages.

There is no concrete timing on the web scrapping plugin yet, but it is in our backlog. If you have any suggestions or use cases for the future plugin, please let us know!

**Pull request closed without merging**